packages/react/src/auto/AutoInput.tsx

```diff
@@ -0,0 +1,19 @@
+import React from "react";
+
+export interface AutoInputComponent<P> extends React.FC<P> {
+  __autoInput: true;
+}
+
+export function autoInput<P extends { field: string }>(Component: React.FC<P>): AutoInputComponent<P> {
+  const WrappedComponent: React.FC<P> = (props) => {
+    return <Component {...props} />;
+  };
+
+  (WrappedComponent as AutoInputComponent<P>).__autoInput = true;
+
+  return WrappedComponent as AutoInputComponent<P>;
+}
+
+export function isAutoInput(component: React.ReactElement): component is React.ReactElement<any, AutoInputComponent<any>> {
+  return typeof component.type === "function" && "__autoInput" in component.type;
+}
```

Comment:
How and where would this function be called?

Reply:
its coming in an upcoming PR